src/commands/reword.rs

```diff
@@ -97,68 +97,154 @@ pub fn reword(
         dump_rebase_constraints,
         dump_rebase_plan,
     } = *rebase_options;
-    let now = SystemTime::now();
-    let event_tx_id = event_log_db.make_transaction_id(now, "reword")?;
-    let pool = ThreadPoolBuilder::new().build()?;
-    let repo_pool = RepoResource::new_pool(&repo)?;
-
-    for subtree_root in subtree_roots.iter() {
-        let base_commit = subtree_root;
-        let base_parents = subtree_root.get_parents();
-        // TODO what about multiple parents?
-        let parent = base_parents.get(0).unwrap();
-        let source_oid: NonZeroOid = base_commit.get_oid();
-        let dest_oid: NonZeroOid = parent.get_oid();
-
-        let rebase_plan = {
-            let mut builder = RebasePlanBuilder::new(&dag);
-            builder.move_subtree(source_oid, dest_oid)?;
-            builder.reword_commits(&messages)?;
-            builder.build(
-                effects,
-                &pool,
-                &repo_pool,
-                &BuildRebasePlanOptions {
-                    dump_rebase_constraints,
-                    dump_rebase_plan,
-                    detect_duplicate_commits_via_patch_id,
-                },
-            )?
-        };
-        let _result = match rebase_plan {
-            Ok(None) => {
-                writeln!(effects.get_output_stream(), "Nothing to do.")?;
-                return Ok(0);
-            }
-            Ok(Some(rebase_plan)) => {
-                let options = ExecuteRebasePlanOptions {
-                    now,
-                    event_tx_id,
-                    preserve_timestamps: get_restack_preserve_timestamps(&repo)?,
-                    force_in_memory,
-                    force_on_disk,
-                    resolve_merge_conflicts,
-                    check_out_commit_options: Default::default(),
-                };
-                execute_rebase_plan(effects, git_run_info, &repo, &rebase_plan, &options)?
-            }
+
+    let subtree_roots_and_plans: Vec<(Commit, Option<RebasePlan>)> = {
+        let pool = ThreadPoolBuilder::new().build()?;
+        let repo_pool = RepoResource::new_pool(&repo)?;
+        let builder = RebasePlanBuilder::new(&dag);
+
+        let subtree_roots_and_plans =
```

Comment:
Ah, you already figured out what I commented in my last review. To simplify this, attach all of the rewords to the same `RebasePlanBuilder` and just execute one rebase plan. `git sync` is different because some of the stacks may fail, and we want to proceed to process other stacks. In this case, none of the rebases should fail.

Reply:
The part about `git sync` being different b/c of possible failures/nothing-to-dos makes sense, but can you clarify what you mean about a single rebase plan? Do you mean that we should be rewriting multiple subtrees w/i a single plan, or that we should instead be tracing the dag back to a single common ancestor and just executing a rebase on that tree to catch all of the requested rewordings? If the former: cool! I hadn't tried that b/c I didn't realize that was possible!; if the latter: I would be concerned that that could potentially encompass a *lot* of commits pretty quickly. Regardless, I look forward to your clarification.

Reply:
Oh, nevermind, I just read your other comment where you spelled it all out. Thanks!

Reply:
Yep, just call `.move_subtree` on the same `RebasePlanBuilder` multiple times. You can pass in unrelated trees or commits which are ancestors of one another and everything should just work.  (It's a pretty nifty API if I do say so myself 😎)

Reply:
The example in restack.rs might be a better guide than the one in sync.rs:

https://github.com/arxanas/git-branchless/blob/45ace45579804697f8daeb551750424bf8e8538b/src/commands/restack.rs#L119-L172

Reply:
Thank you, `git restack` was helpful. And yes, the API is slick. The hardest part for me has just been getting up to speed w/ the language and figuring out the idioms. After that, the feature basically wrote itself. 😄